bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java

```diff
@@ -49,16 +56,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String usernameInput = parameters.get(USERNAME)[0];
     String emailInput = parameters.get(EMAIL)[0];
     String addressInput = parameters.get(ADDRESS)[0];
-    String pictureInput = parameters.get(PICTURE)[0];
+    String blobKeyString = getUploadedFileUrl(request, PICTURE);
     String phoneInput = parameters.get(PHONE)[0];
     String notesInput = parameters.get(NOTES)[0];
 
     ObjectifyService.ofy()
         .save()
         .entity(new Request(requestId, userId, nameInput, usernameInput, emailInput, addressInput,
-            pictureInput, phoneInput, notesInput))
+            blobKeyString, phoneInput, notesInput))
         .now();
 
     response.sendRedirect("/success.jsp");
   }
+
+  /** Gets the URL of the file the user uploaded and checks for no file uploaded */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
```

Comment:
Might want to do some error handling? Or at least let the user know that their image wasn't saved properly? Same w live server section